src/firmware_update/rockusb_fwudate.rs

```diff
@@ -26,10 +26,32 @@ use rusb::GlobalContext;
 use std::{mem::size_of, ops::Range, time::Duration};
 use tokio::sync::mpsc::Sender;
 
+pub enum Chip {
+    RK3566,
+    RK3588,
+}
+
+impl Chip {
+    pub const fn vid_pid(&self) -> (u16, u16) {
+        match self {
+            Chip::RK3566 => (0x2207, 0x350a),
+            Chip::RK3588 => (0x2207, 0x350b),
+        }
+    }
+
+    const fn loader(&self) -> &'static [u8] {
+        match self {
+            Chip::RK3566 => SPL_LOADER_RK3566,
+            Chip::RK3588 => SPL_LOADER_RK3588,
+        }
+    }
+}
+
+const SPL_LOADER_RK3566: &[u8] = include_bytes!("./rk356x_spl_loader_v1.18.112.bin");
```

Comment:
I built this file from https://github.com/rockchip-linux/rkbin like this:

```sh
tools/boot_merger RKBOOT/RK3566MINIALL.ini
```

Reply:
Is it possible to make a `build.rs` script that builds this loader? Checking in binaries is not really great to maintain. I can see if i can do the same for the other loader

Reply:
I think we could, but I'm not sure how valuable it would be. The `rkbin` repo is itself just a bunch of checked-in binary blobs, a couple of which get smashed together by their magic `boot_merger` binary tool. It's all pretty opaque.

In these situations I like to ensure that the build of the committed binary is _repeatable_ and _automatable_ so that updates are easy, but to commit the final binary to the repo as a matter of practicality. Pulling the entire rkbin repo as a submodule on every build is pretty heavy...